lib/zinnia.js

```diff
@@ -121,36 +126,45 @@ export async function run ({
     }
   }
 
-  const childProcess = execa(zinniadExe, modules, {
-    cwd: moduleBinaries,
-    env: {
-      FIL_WALLET_ADDRESS,
-      STATE_ROOT,
-      CACHE_ROOT
-    }
-  })
+  const childProcesses = []
+  let exitReason
 
-  childProcess.stdout.setEncoding('utf-8')
-  childProcess.stdout.on('data', data => {
-    handleEvents({
-      contract,
-      ethAddress,
-      onActivity,
-      onMetrics,
-      text: data
-    }).catch(err => {
-      console.error(err)
-      Sentry.captureException(err)
+  for (const { module } of ZINNIA_MODULES) {
+    if (MODULE_FILTER !== '' && module !== MODULE_FILTER) continue
+
+    // all paths are relative to `moduleBinaries`
+    const childProcess = execa(zinniadExe, [`${module}/main.js`], {
+      cwd: moduleBinaries,
+      env: {
+        FIL_WALLET_ADDRESS,
+        STATE_ROOT,
+        CACHE_ROOT
+      }
     })
-  })
-  childProcess.stderr.pipe(process.stderr, { end: false })
+    childProcesses.push(childProcess)
 
-  let exitReason
-  childProcess.on('exit', (code, signal) => {
-    exitReason = signal ? `via signal ${signal}` : `with code: ${code}`
-    const msg = `Zinnia exited ${exitReason}`
-    onActivity({ type: 'info', message: msg })
-  })
+    childProcess.stdout.setEncoding('utf-8')
+    childProcess.stdout.on('data', data => {
+      handleEvents({
+        module,
+        contract,
+        ethAddress,
+        onActivity,
+        onMetrics,
+        text: data
+      }).catch(err => {
+        console.error(err)
+        Sentry.captureException(err)
+      })
+    })
+    childProcess.stderr.pipe(process.stderr, { end: false })
+
+    childProcess.on('exit', (code, signal) => {
+      exitReason = signal ? `via signal ${signal}` : `with code: ${code}`
+      const msg = `Zinnia exited ${exitReason}`
```

Comment:
There will be multiple Zinnia processes, one per module. This message will be difficult to troubleshoot because we don't know which module exited with the reported reason.

I think we should change all messages and rename `Zinnia` to the module name (`Spark` or `Voyager`.)

Reply:
Good idea! Done in https://github.com/filecoin-station/core/pull/323/commits/22b89c79a1e7d8f802dce7db385c59bfa748e0d8 (forgive the busy commit)